### PR TITLE
Gameplay

### PR DESCRIPTION
**Pull request by Nopey**
Implemented gameplay.
Tape XOR, i think this is.
Killing a friend applies XOR, killing an enemy applies OR.

### CODE REVIEW

File: Moveset.hpp [hunk 1]
```diff
@@ -34,10 +34,49 @@ class Moveset {
...
+    ret.vertical = (this->vertical ^ m.vertical);
+    ret.diagonal = (this->diagonal ^ m.diagonal);
+    ret.circular = (this->circular ^ m.circular);
+
+    
+    // If right side has an offet ret doesn't push it back to ret's offsets
+    for (auto a : offsets) {
+      for ( auto b : m.offsets ) {
+      	if ( a == b )
+          //Will they notice?
```
Comment (benoxoft) [inline Moveset.hpp]: yes

File: Moveset.hpp [hunk 2]
```diff
@@ -34,10 +34,49 @@ class Moveset {
...
+          goto continueOuterLoop;
+      }
+      ret.offsets.push_back(a);
+      continueOuterLoop:
+      continue;
+    }
+    for (auto a : m.offsets) {
+      for ( auto b : offsets ) {
+      	if ( a == b )
+          goto continueSecondOuterLoop;
```
Comment (benoxoft) [inline Moveset.hpp]: 🙄

Comment (Nopey) [reply Moveset.hpp]: What would you suggest?

Comment (Nopey) [reply Moveset.hpp]: Ignoring the stigma of the GOTO, it is performing exactly the same as a CONTINUE here.
Its just a named continue. :eyes:

File: Board.cpp [hunk 3]
```diff
@@ -71,14 +72,43 @@ Board::Board(Game &game, sf::Vector2f position,
...
+      if(y < 2){
+        player=false;
+      }else if(boardSize-y-1 < 2){
+        player=true;
+      }else{
+        pieces.push_back(nullptr);
+        continue;
+      }
+      pieces.push_back(new Piece(game, *this,
+        (y%(boardSize-1)) ? Basepiece::PAWN : CHESS_ORDER[player?x:(boardSize-x-1)],
```
Comment (benoxoft) [inline Board.cpp:102]: I don't think this change make the code easier to understand. It's not clear that this piece of code is where the pieces are placed on the board. Maybe add a comment or put it in its own method?

File: TestState.cpp [hunk 4]
```diff
@@ -35,30 +33,19 @@ void TestState::onEnter(Game &game) {
...
+  board.onEvent(event);
 }
 
 std::shared_ptr<State> TestState::update(sf::RenderWindow& window) {
 
   board.update(window);
 
-  for(auto &piece : pieces.playerPieces) {
-    piece.update(window, board);
+  for(auto &piece : board.pieces) {
```
Comment (benoxoft) [inline TestState.cpp:46]: now that the enemy pieces and player pieces are combined, I can move both set of the pieces. We will have to be able to know who is the owner of a piece.

File: Piece.cpp [hunk 5]
```diff
@@ -12,43 +12,89 @@
...
+      moveset.offsets.push_back(sf::Vector2i(-1, 0));
+      moveset.offsets.push_back(sf::Vector2i( 0,-1));
+      break;
+    case Basepiece::QUEEN:
+      moveset.diagonal = true;
+      moveset.horizontal = true;
+      moveset.vertical = true;
+      break;
+    case Basepiece::PAWN:
+      moveset.offsets.push_back(sf::Vector2i( 0, player ? -1 : 1));
```
Comment (benoxoft) [inline Piece.cpp:33]: thinking about it, pawns can move diagonally to kill an enemy piece. I'm wondering how this will be handled @sweetsbeats @Nopey

**Review by benoxoft — APPROVED** [review]
👍

File: TestState.cpp (hunk 4 above)
Comment (Nopey) [reply TestState.cpp:46]: On the Piece class there is a `player` integer

File: Piece.cpp (hunk 5 above)
Comment (Nopey) [reply Piece.cpp:33]: There is a lot of missing functionality from our chess, including castling, promotion, the weird pawn double move thing i never understood..
My recommendation is to not worry about it

**Pull request merged**